### PR TITLE
Add 'extending verbosity' sub-section

### PR DESCRIPTION
**Pull request by mpadge**

### CODE REVIEW

File: content/blog/2024-02-06-verbosity-control-packages/index.Rmd [hunk 1]
```diff
@@ -3,7 +3,7 @@ title: Please Shut Up! Verbosity Control in Packages
 author: 
 - Mark Padgham
 - Maëlle Salmon
-date: '2024-01-26'
+date: '2024-0w-06'
```
Comment (maelle) [inline content/blog/2024-02-06-verbosity-control-packages/index.Rmd:6]:
```suggestion
date: '2024-02-06'
```

File: content/blog/2024-02-06-verbosity-control-packages/index.Rmd [hunk 2]
```diff
@@ -140,6 +141,40 @@ pkg_message("reawakened message")
...
 And my local package-level verbosity options can then override the rlang verbosity controls.
 
+### Extending levels of verbosity
+
+Finally, we'll briefly demonstrate how the rlang/cli two-level verbosity control of "quiet"/"verbose" can be extended to implement a "debug" level in your own package.
+This again requires a custom message handler, slightly modified from the above to issue debug messages only:
+
+```{r}
+debug_msg <- function(...) {
+  msg <- getOption("mypackage.verbose", "quiet") == "debug"
```
Comment (maelle) [inline content/blog/2024-02-06-verbosity-control-packages/index.Rmd:151]:
```suggestion
  msg <- (getOption("mypackage.verbose", "quiet") == "debug")
```

I find this clearer even if the parentheses are in practice useless

Comment (maelle) [reply content/blog/2024-02-06-verbosity-control-packages/index.Rmd:151]: why not call msg something informative, "is_debugging" or so?

File: content/blog/2024-02-06-verbosity-control-packages/index.Rmd [hunk 3]
```diff
@@ -140,6 +141,40 @@ pkg_message("reawakened message")
 
 And my local package-level verbosity options can then override the rlang verbosity controls.
 
+### Extending levels of verbosity
+
+Finally, we'll briefly demonstrate how the rlang/cli two-level verbosity control of "quiet"/"verbose" can be extended to implement a "debug" level in your own package.
+This again requires a custom message handler, slightly modified from the above to issue debug messages only:
+
+```{r}
+debug_msg <- function(...) {
```
Comment (maelle) [inline content/blog/2024-02-06-verbosity-control-packages/index.Rmd:150]: I'd find the example more convincing if you demo-ed it within another function say.

```r
my_function <- function(x, ...) {
# do stuff
debug_msg(x = x, ...)
# more stuff
}

```

**Review by maelle — APPROVED** [review]
Thanks!! A few comments

File: content/blog/2024-02-06-verbosity-control-packages/index.Rmd (hunk 2 above)
Comment (mpadge) [reply content/blog/2024-02-06-verbosity-control-packages/index.Rmd]: Good idea - i'll do that right now and ping you on commit

File: content/blog/2024-02-06-verbosity-control-packages/index.Rmd (hunk 3 above)
Comment (mpadge) [reply content/blog/2024-02-06-verbosity-control-packages/index.Rmd:150]: I'm not entirely sure what you mean, so can you please do that? Happy to review if you want

Comment (maelle) [reply content/blog/2024-02-06-verbosity-control-packages/index.Rmd:150]: your `debug_msg()` is supposed to be used within the package functions, so why not demo it used within package functions? but also fine to drop

Comment (mpadge) [reply content/blog/2024-02-06-verbosity-control-packages/index.Rmd:150]: Oh yeah, right - just let me do that, then i reckon i'm done and you can merge this

Comment (maelle) [reply content/blog/2024-02-06-verbosity-control-packages/index.Rmd:150]: you can also merge since I approved it :wink:

**Comment by mpadge** [comment]
@maelle That should do it now. Happy for you to merge and publish. Thank you!

**Pull request merged**